### DISCUSSION TITLE
Some problems with PyQt5 when running the modelOrderReduction.py

### DISCUSSION THREAD

**Discussion by tolstoys19** [sofa-framework/sofa#4578] [ModelOrderReduction] [answered]
Hello

I have been trying to install the MOR plugin on ubuntu. I have previously installed it succesfully on Windows but right now in ubuntu I am getting some issued when running the modelOrderReduction.py from the tools folder on the plugin. 

When running phase 1 I get the following message:

![image](https://github.com/sofa-framework/sofa/assets/84059736/fe9aaa58-8d40-4982-9942-b71420514e6e)

Then when I check the some of the phase1_snapshots.py I get the following message

![image](https://github.com/sofa-framework/sofa/assets/84059736/275cf333-f2c9-4813-8dd1-b41de29ddff3)

I am directly using the modelOrderReduction.py provided on the tools folder of the master branch for linux. I am trying to just do the example of the diamondRobot.

@VannesteFelix I tried to send you a message about this through the gitter before noticing that we migrated from there. I am very sorry about that. Do you by any chance know what might be the issue?

**Comment by tolstoys19** [comment]
I forgot to mention that I am running python3.10  on a virtual environment on miniconda.

**Comment by bakpaul** [comment]
SOFA binaries have been compilesd with Qt5.12.12. That may be the failure reason. Maybe install Qt5.12.12 in your cnoda env instead of 5.15.
This is how we do it on our builders (using pip) : 
`python3.10 -m pip install aqtinstall && python3.10 -m aqt install-qt --outputdir $QT_INSTALLDIR linux desktop 5.12.12 gcc_64 -m qtcharts qtwebengine`

**Reply by tolstoys19** [reply]
Thank you for your quick reply. I tried your solution but I still get the exact same error message. Could it be a problem with the version of pyqt5? I currently can only install pyqt5 version 5.15. I get an error message when I try to install pyqt5 version 5.12.
![image](https://github.com/sofa-framework/sofa/assets/84059736/ec7b008f-1b5d-4f6d-b946-e3849e3c3085)

**Reply by bakpaul** [reply]
I looked at pyqt5_sip versions it is strange because the requirements could be met with versions `4.19.15` to `4.19.19`
Maybe, first thing I would do is to, in addition to the uninstall of the version 5.15, purge all qt packages in your pip environment, reboot your computer and retry installing this version.

A second option if this doesn't work, try passing through `aqt` as we do in our builders.

**Reply by tolstoys19** [reply, marked as answer]
Thank you for the help. I removed all instances of qt in my environment but I was still getting the same problem when reinstalling. 
I was able to solve  the problem by specifying the version of 5.12(I didn't need to do this for 5.15) with:

pip install PyQt5==5.12.2 

Got the solution from one of the comments in here: https://stackoverflow.com/questions/56418874/cant-install-pyqt5-using-pip
